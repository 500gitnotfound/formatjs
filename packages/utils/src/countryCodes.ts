import alpha3CountryCodes from './iso3166Alpha3CountryCodes.json'

const COUNTRY_CODE_ALPHA2 = new Set(
  Object.keys(alpha3CountryCodes).map(
    key => alpha3CountryCodes[key as 'USA'] as string
  )
)

export function countryCodeAlpha3ToAlpha2(alpha3?: string): string | undefined {
  if (!alpha3) {
    return
  }
  const upper = alpha3.toUpperCase()
  // Lenient here, if it's already a 2 letter code, just return it
  if (alpha3.length === 2) {
    return COUNTRY_CODE_ALPHA2.has(upper) ? upper : undefined
  }
  return alpha3CountryCodes[upper as 'USA']
}
