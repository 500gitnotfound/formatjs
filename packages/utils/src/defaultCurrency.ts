import {canonicalizeCountryCode} from './countryCodes'
import * as data from './defaultCurrencyData.generated.json'

/**
 * Look up default currency for a country code.
 * @param countryCode country code (alpha-2)
 * @returns default currency code, or USD if not found
 */
export function defaultCurrency(countryCode?: string): string | undefined {
  countryCode = canonicalizeCountryCode(countryCode)
  return (
    (countryCode && countryCode in data && data[countryCode as 'US']) || 'USD'
  )
}
