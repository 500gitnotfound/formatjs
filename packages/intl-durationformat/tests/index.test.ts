import {DurationFormat} from '../src/core'
DurationFormat.__addLocaleData({
  data: {
    digitalFormat: {
      latn: ':',
    },
    nu: ['latn'],
  },
  locale: 'en',
})
test.skip('Intl.DurationFormat', function () {
  expect(new DurationFormat('en').format({years: 1})).toBe('1 year')
})
