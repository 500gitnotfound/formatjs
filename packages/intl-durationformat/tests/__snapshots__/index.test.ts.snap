// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intl.DurationFormat resolvedOptions 1`] = `
{
  "days": "short",
  "daysDisplay": "auto",
  "fractionalDigits": undefined,
  "hours": "short",
  "hoursDisplay": "auto",
  "locale": "en",
  "microseconds": "short",
  "microsecondsDisplay": "auto",
  "milliseconds": "short",
  "millisecondsDisplay": "auto",
  "minutes": "short",
  "minutesDisplay": "auto",
  "months": "short",
  "monthsDisplay": "auto",
  "nanoseconds": "short",
  "nanosecondsDisplay": "auto",
  "numberingSystem": "latn",
  "seconds": "short",
  "secondsDisplay": "auto",
  "style": "short",
  "weeks": "short",
  "weeksDisplay": "auto",
  "years": "short",
  "yearsDisplay": "auto",
}
`;
